shap/plots/_text.py

```diff
@@ -722,90 +769,206 @@ def get_colors(shap_values):
 def heatmap(shap_values):
     
     uuid = ''.join(random.choices(string.ascii_lowercase, k=20))
-    
+
     def get_color(shap_value,cmax):
         scaled_value = 0.5 + 0.5 * shap_value / cmax
         color = colors.red_transparent_blue(scaled_value)
         color = (color[0]*255, color[1]*255, color[2]*255, color[3])
         return color
     
-    
+    def generate_tree(shap_values):
+        num_tokens = shap_values.data.shape[0]
+        token_list = {}
+        max_vals = np.zeros(shap_values.values.shape[1])
+
+        for index in range(num_tokens):
+            node_content = {}
+            node_content['tokens'] = shap_values.data[index]
+            node_content['children'] = {}
+            token_list[str(index)] = node_content
+
+        counter = num_tokens
+        for pair in shap_values.clustering:
+            first_node = str(int(pair[0]))
+            second_node = str(int(pair[1]))
+
+            new_node_content = {}
+
+            new_node_content['tokens'] = token_list[first_node]['tokens'] + token_list[second_node]['tokens']
```

Comment:
nit: 'tokens' and other repeating strings to constants

Reply:
Moved to constants